### PR TITLE
Retry all ClientErrors and log retries

### PR DESCRIPTION
**Pull request by apurvis**
sorry for another PR but `Faraday::Error::ClientError` is [a superclass of all the errors i am seeing](https://github.com/lostisland/faraday/blob/80c0e66102168939df04682573bc267e089a43ae/lib/faraday/error.rb#L46-L57) out of SG these days... i wrote support because something seems wrong with their servers but i'm hoping this will at least make it less of a problem

(i actually saw a `ConnectionFailed` exception just in the few minutes since i patched the parser error retries)

@jarthod

### CODE REVIEW

File: lib/survey_gizmo/connection.rb [hunk 1]
```diff
@@ -54,12 +54,11 @@ def retriable_args
           interval: SurveyGizmo.configuration.retry_interval,
           tries: SurveyGizmo.configuration.retry_attempts + 1,
           on: [
-            SurveyGizmo::BadResponseError,
-            SurveyGizmo::RateLimitExceededError,
             Errno::ETIMEDOUT,
+            Faraday::Error::ClientError,
```
Comment (apurvis) [inline lib/survey_gizmo/connection.rb]: i added this; other changes are just alphabetizing

**Comment by apurvis** [comment]
Added the logging of retries to this PR

**Comment by apurvis** [comment]
also added basic log instrumentation for how long it takes for SG to respond to a request.

**Comment by jarthod** [comment]
Cool, released `6.2.9` ☺

**Pull request merged**